prefill_haalcentraalhr/tests/test_plugin_with_token_exchange.py

```diff
@@ -0,0 +1,103 @@
+import json
+from dataclasses import dataclass
+from pathlib import Path
+from unittest.mock import patch
+
+from django.core.cache import cache
+from django.test import TestCase, override_settings
+
+from openforms.authentication.constants import AuthAttribute
+from openforms.pre_requests.base import PreRequestHookBase
+from openforms.pre_requests.registry import Registry
+from openforms.registrations.contrib.zgw_apis.tests.factories import ServiceFactory
+from openforms.submissions.models import Submission
+from openforms.submissions.tests.factories import SubmissionFactory
+from requests.auth import AuthBase
+from requests_mock import Mocker
+from zgw_consumers.constants import APITypes
+from zgw_consumers.test import mock_service_oas_get
+
+from ..models import HaalCentraalHRConfig
+from ..plugin import HaalCentraalHRPrefill
+
+FILES_DIR = Path(__file__).parent / "files"
+
+register = Registry()
+
+
+@dataclass
+class TestAuth(AuthBase):
+    submission: Submission
+
+    def __call__(self, request):
+        request.headers["Authorization"] = "EXCHANGED TOKEN VALUE"
+        return request
+
+
+@register("token-exchange-test")
+class TokenExchangePreRequestHookTest(PreRequestHookBase):
+    def __call__(
+        self,
+        method,
+        url,
+        kwargs,
+        context,
+    ):
+        kwargs["auth"] = TestAuth(submission=context["submission"])
+        return
+
+
+@Mocker()
+@override_settings(ZGW_CONSUMERS_TEST_SCHEMA_DIRS=[FILES_DIR])
+class HaalCentraalHRPluginWithTokenExchangeTests(TestCase):
+    def test_token_exchange(self, m):
+        mock_service_oas_get(
+            m,
+            url="http://haalcentraal-hr.nl/api/",
+            service="haalcentraal-hr-oas",
+            oas_url="https://haalcentraal-hr.nl/api/schema/openapi.yaml",
+        )
+
+        with open(FILES_DIR / "maatschapplelijkeactiviteiten-response.json", "rb") as f:
```

Comment:
Only noticed now (it may be deliberate ;) ) but there's a typo:

```suggestion
        with open(FILES_DIR / "maatschappelijkeactiviteiten-response.json", "rb") as f:
```
the current form looks like "maatsUGLYactiviteiten-response.json" :laughing:

Reply:
ouuuups :laughing: